package/upgrade/lib.sh

```diff
@@ -242,17 +242,17 @@ install_addon()
   local name=$1
   local namespace=$2
 
-  kubectl apply -f ./addons/${name}.yaml -n $namespace
+  kubectl apply -f /usr/local/share/addons/${name}.yaml -n $namespace
 }
 
 wait_for_addons_crd()
 {
-  item_count=$(kubectl get customresourcedefinitions addons.harvesterhci.io -o  jsonpath='{.metadata.name}')
+  item_count=$(kubectl get customresourcedefinitions addons.harvesterhci.io -o  jsonpath='{.metadata.name}' || true)
   while [ -z "$item_count" ]
   do
     echo "wait for addons.harvesterhci.io crd to be created"
     sleep 10
-    item_count=$(kubectl get customresourcedefinitions addons.harvesterhci.io -o  jsonpath='{.metadata.name}')
+    item_count=$(kubectl get customresourcedefinitions addons.harvesterhci.io -o  jsonpath='{.metadata.name}' || true)
```

Comment:
Is here waiting form CRD to be applied by `kubectl` ?  should it be `while [ ! -z "$item_count" ]` ?

Reply:
i think its correct, the item_count will not be set until the harvester controller has deployed the crd

Reply:
em, it's correct.

Reply:
maybe should loop at most certain times and then break to report error, avoid dead loop